orchestra/cmds/components.py

```diff
@@ -70,21 +74,79 @@ def handle_components(args):
             if not fnmatch(branch, args.branch):
                 continue
 
+        # Filter by install status
+        metadata = get_installed_metadata(component_name, config)
+        installed_build = metadata and metadata.get("build_name")
+
+        if args.installed and installed_build is None \
+                or args.not_installed and installed_build is not None:
+            continue
+
+        components_to_print.add(component)
+
+    if args.json:
+        print_json(components_to_print, config)
+    else:
+        print_human_readable(components_to_print, config, args)
+
+
+def print_json(components, config):
```

Comment:
I'm thinking: the constructor of `Component` and `Build` now takes *json* object, am I right?
Shall we have a `Component.toJSON` and `Build.toJSON` methods compatible with the constructor?
The following code belongs to the `Component`/`Build` classes.

Reply:
Yes, they now take a parsed JSON, that is, a dictionary with a non trivial structure.
We can write those methods, however nobody would use them at the moment, so would it make sense to have them?

Reply:
They just belong there from a sweng POV.

Reply:
Done! I called the method `serialize` because it returns a `dict` which is trivially serializable as JSON or YAML or whatever format the caller needs.

Reply:
Can we have a test that ensures that building a `Component` (or a `Build`) from the object obtained by the `serialize` method is the same as the original one? Otherwise I feel like we'll soon lose this property accidentally.

```python
assert Component(mycomponent.serialize()) == mycomponent
```

Reply:
Done! At first I was not happy about this because I thought I would need to implement `__eq__` for `Component`, `Build` and `Action`, which are not comparable by design (*). Then I realized we could just compare the hashes.

(*) Rationale:
- there's no reason to have multiple instances of equivalent objects
- `__eq__` or a similar comparator function would be more code that needs to be maintained, tested, and could break